hs_core/page_processors.py

```diff
@@ -34,6 +35,16 @@ def get_page_context(page, user, resource_edit=False, extended_metadata_layout=N
 
     metadata_status = _get_metadata_status(content_model)
 
+    relevant_tools = []
+    for tool in ToolResource.objects.all():
+        if len(tool.metadata.res_types.all())>0:
```

Comment:
I took out the unnecessary conditional. However, I'm not sure what you mean by making it one query.